### PR TITLE
Call assignment details from new route after a route change.

### LINKED ISSUE #5664: Assignment view does not update when clicking banner link
If an assignment page is displayed, e.g. /assignments/1 with a banner link pointing to /assignments/2 clicking the link will not update assignment details.

Adding this code to [https://github.com/OpenSlides/OpenSlides/blob/master/client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts](url) seems to fix the problem:

```
    private navigationSubscription: Subscription;

    public ngOnInit(): void {
        this.observeRoute();

    public ngOnDestroy(): void {
        if (this.navigationSubscription) {
            this.navigationSubscription.unsubscribe();
        }
        super.ngOnDestroy();
    }

    public observeRoute(): void {
        this.navigationSubscription = this.router.events.subscribe(navEvent => {
            if (navEvent instanceof NavigationEnd) {
                this.cleanSubjects();
                this.getAssignmentByUrl();
            }
        });
    }
```

### PR DESCRIPTION
**Pull request by jwinzer**
This fixes #5664

### CODE REVIEW

File: client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts [hunk 1]
```diff
@@ -210,12 +217,36 @@ export class AssignmentDetailComponent extends BaseViewComponentDirective implem
...
+            this.navigationSubscription.unsubscribe();
+        }
+        super.ngOnDestroy();
+    }
+
+    /**
+     * Observes the route for events. Calls to clean all subs if the route changes.
+     * Calls the assignment details from the new route.
+     */
+    public observeRoute(): void {
```
Comment (tsiegleauq) [inline client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts:241]: should be private

**Review by tsiegleauq — CHANGES_REQUESTED** [review]

File: client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts (hunk 1 above)
Comment (jwinzer) [reply client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts:241]: Makes sense. I blindly copied this from 
[motion-detail](https://github.com/OpenSlides/OpenSlides/blob/master/client/src/app/site/motions/modules/motion-detail/components/motion-detail/motion-detail.component.ts#L609)

Comment (tsiegleauq) [reply client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts:241]: That's what I thought. Should also be private there.

Comment (jwinzer) [reply client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.ts]: Commit has been added.

**Review by tsiegleauq — APPROVED** [review]
Looks good. Please rebase to one commit

**Comment by jwinzer** [comment]
Done.

**Review by emanuelschuetze — APPROVED** [review]

**Pull request merged**